### ISSUE TITLE
Add Signed-off-by line in commit log message.

### ISSUE THREAD

**Issue by pvital** [atom/github#914]
### Description

Make possible to add a Signed-off-by line by the committer at the end of the commit log message. Many projects, even hosted in GitHub, needs a signoff line to prove the authorship of the commit.

When running ```$ git commit -s <file> ``` the default editor is opened to write the commit log message and the Signed-off-by line came inserted automatically.

**Expected behavior:**

Enable the option to signoff the commit, by a selectiong a check box on UI near the "Commit"  button (like the "Amend" option), and then have the ```Signed-off-by: <committer-email>``` line in the end of the field "Commit message".

**Actual behavior:**

No option to signoff a commit.

**Reproduces how often:**

100% of the time I want commit using github package.

### Versions

OS: Linux, Fedora 25

```
$ atom-beta --version
Atom    : 1.18.0-beta2
Electron: 1.3.15
Chrome  : 52.0.2743.82
Node    : 6.5.0

$ apm-beta --version
apm  1.18.1
npm  3.10.10
node 6.9.5 x64
python 2.7.13
git 2.9.4
```

**Comment by kfrz** [comment]
Yeah, it would be nice to have an option to point to a key and have a check box for "sign commit"

**Comment by mikebrow** [comment]
yeah signing is a must requirement for just about every commit these days

**Comment by lnxg33k** [comment]
This is a nice important feature to include.

**Comment by Arcanemagus** [comment]
@kfrz `-s`, which is what this issue is talking about, extends to `--signoff`, you are talking about `-S` or `--gpg-sign` which is an entirely different option 😉.

Note that you can enable signing for all commits for a project with:
```sh
git config --local commit.gpgsign true
```

Change `--local` to `--global` to enable for all projects if you want that instead. Note that if you have it enabled globally and explicitly set it to false in a project locally just that project will not ask for commit GPG signatures.

Atom supports signing your commits if you have told `git` to do so in the config, since it's just shelling out to `git` 😉.

**Comment by claell** [comment]
@Arcanemagus 
> Atom supports signing your commits if you have told git to do so in the config, since it's just shelling out to git.

The problem is I don't know how to tell it to git in the config. There does not seem to be a setting for that, right?

**Comment by Arcanemagus** [comment]
@claell Reading your comments in the linked issue, you are talking about the issue in this PR. My comment was about GPG signing your commits, which is an entirely different subject.

**Comment by claell** [comment]
Ah sorry. Thought the last sentence of your comment was about the signoff again and not the GPG signing.

I did not find much confirmation on the link to stackoverflow you provided that the intent is only for patches, not for commits.

**Comment by mikebrow** [comment]
Not sure why my comment was marked as off-topic. Signing a commit using the lowercase "-s" to inject your signed off by email in a commit is a ubiquitous use of github. Just about every major github project requires signify. Thus this feature not being available in Atom forces users of Atom to commit outside the editor, making commit in the editor pretty much a useless feature. The importance of this feature was not specified in the issue. There is no feature in git for automatically signing (-s) commits. If a user accidentally uses Atom to commit they will have to amend their commit by signing it outside Atom and force pushing the amended commit.

**Comment by ValZapod** [comment]
> `git` only provides a method for automatically adding `--signoff` for _patches_,

What about git send-email -1??? It needs sign-off-by too! WTF.
